dannielRolfe/file-1.js

```diff
@@ -0,0 +1,255 @@
+
+
+//i) 
+typeof x + 1 === "number" /*Output: always false because the operand "===" checks for equal value and in this operation "number" which is a string can not be of equal value to x + 1*/
+
+//j) 
+
+typeof x % 2 === "number" /*Output: always false because the operand "===" checks for equal value and in this operation "number" which is a string can not be of equal value to x mod 2 in any givin situation*/
+
+//k) 
+
+typeof (x % 2) === "number" /*Output: always true because the order of operations is to first calculate the mod in parenthesis and then check for equality. Therefore, if x is equal to 1 (1%2) === "number", 1 === "number":true*/
+
+//l) 
+
+x++ == ++x /*Output: always false because x++ returns x, then increments it and ++x increments x, and then returns it. Therefor, if x equals on 1++ == ++1 would evaluate to 1++ == 3 which is false. */ 
+var x =1;
+++x == x++
+1 == 3
+false
+//m) 
+
+++x == x++ /*Output: always true because ++x increments x, and then returns it. x++ returns x, then increments it. Therefore: */
+var x = 1;
+++x == x++
+2==2
+true
+
+//n) ---- incorrect 
+"1"+x == 1+x /*Output: false because if x=1, "1"+x will evaluate to 11 and 1 + 1 will evaluate to 2. Therefore, x=1; "1"+1 == 1+x; 11 == 2; false*/
+
+/*correct answer - true when x is any string:
+ '1'+x == 1+x
+    '1'+'****' == 1+x
+    '1****' == 1+x
+    '1****' == '1****' */
+
+//o) 
+"0"+1 == 1 /*Output: true because "0"+1 will evaluate to 1, and 1 == 1 is true*/
+
+//p) 
+(typeof (x+1))===(typeof x) /*Output: true with the exception of when x is equal to null, undefined, or boolean */
+
+//q) 
+(x*1 == x) || ((typeof x) != "number") /*Output: true because when x equals 1 this evaluates to (1*1 == 1) true || ((typeof 1) != "number") false. The|| denotes logical or. It produces true if either of the values given to it is true.*/ 
+
+/* side note true unless x is Nan*/
+
+//r) ------------- not correct! 
+(x=(typeof (x+(typeof x))))==x /*Output: Always true*/
+x=1;
+(1=(typeof (1+(typeof 1))))==1
+(1=(typeof (1+("number"))))==1
+(1=("number"))==1
+
+/* Correct answer 
+ true.  For any x, this becomes:
+    (x=(typeof (x+'*****')))==x
+    (x=(typeof 'x*****'))==x
+    (x='string') == x
+    'string' == 'string' */
+
+//Excercise #2
+
+//a)
+var y = 2,x = 4,z = 6,total = 3;
+var calc = ((y+x+z)/total);
+console.log(calc);
+
+//b)  not correct 
+var y = (calc/2), x = (calc/2), z = (calc/2), total = total;
+var calc = ((y+x+z)/total);
+console.log(calc);
+
+/* Correct answer 
+var mean  = (x+y+z/3);
+x = (x+mean)/2;
+y = (y+mean)/2;
+z = (z+mean)/2;
+
+or 
+
+x+= (mean-x)/2;
+y+= (mean-y)/2;
+z+= (mean-z)/2;
+
+*/
+
+
+//Excercise #3
+//a) calculating the area of the rectangle 
+var t = 20
+var b = 2; 
+var l = 2; 
+var r = 12;
+
+var h = (t - b); 
+var w = (r - l);
+console.log(h && w);
+var a = (h*w);
+console.log("The area of the rectangle is " + a);
```

Comment:
In general I'd suggest using more human-readable variable names--`area` instead of `a`, and so on. This will make it much easier to parse your code, and can even make it easier to spot logical errors.